src/app/Library/CrudPanel/Traits/Create.php

```diff
@@ -131,34 +132,75 @@ private function createRelationsForItem($item, $formattedRelations)
                 case 'BelongsToMany':
                 case 'MorphToMany':
                     $values = $relationDetails['values'][$relationMethod] ?? [];
-                    $values = is_string($values) ? json_decode($values, true) : $values;
+                    $values = is_string($values) ? (json_decode($values, true) ?? []) : $values;
+                    $field = $relationDetails['crudFields'][0] ?? [];
 
-                    // disabling ConvertEmptyStringsToNull middleware may return null from json_decode() if an empty string is used.
-                    // we need to make sure no null value can go foward so we reassure that values is not null after json_decode()
-                    $values = $values ?? [];
+                    // if the values are multidimensional, we have additional pivot data.
+                    if (is_array($values) && is_multidimensional_array($values)) {
+                        // if the field allow duplicated pivots, we can't use sync or attach from laravel, we need to manually handle the pivot data.
+                        if ($field['allow_duplicate_pivots'] ?? false) {
+                            $keyName = $field['pivot_key_name'] ?? 'id';
+                            $sentIds = array_filter(array_column($values, $keyName));
```

Comment:
if we get no `$sentIds` then there is no point in filtering for deletion since there are no old values

Reply:
if we get no sent ids, we will perform the array_filter on an empty array .. that's **not expensive**, isn't worth the trouble to don't call it if the array is empty.